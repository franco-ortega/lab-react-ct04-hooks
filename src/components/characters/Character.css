.Character {
    border-radius: 10px;
    border: 'solid';
    margin: '5px';
    padding: '5px';
    width: 'fit-content'
}