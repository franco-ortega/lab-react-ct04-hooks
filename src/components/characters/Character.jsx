import React from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';

export default function Character({ name, japaneseName, image }) {
  return (
    <div
      className={styles.Character}
      style={
        { border: 'solid', margin: '5px', padding: '5px', width: 'fit-content' }
      }
    >
      <p>
        {name}
      </p>
      <p>
        {japaneseName}
      </p>
      <img src={image} alt={name} style={{ height: '100px' }} />
    </div>
  );
}

Character.propTypes = {
  name: PropTypes.string.isRequired,
  japaneseName: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};
