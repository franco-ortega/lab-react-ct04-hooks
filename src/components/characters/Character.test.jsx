import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  // afterEach(() => cleanup());
  it('renders Character', () => {
    const { asFragment } = render(<Character
      name="a title"
      japaneseName="this article"
      image="that image url" />);
    expect(asFragment()).toMatchSnapshot();
  });
});