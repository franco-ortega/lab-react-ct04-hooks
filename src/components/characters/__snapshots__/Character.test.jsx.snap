// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<DocumentFragment>
  <div
    class="Character"
    style="border: solid; margin: 5px; padding: 5px;"
  >
    <p>
      a title
    </p>
    <p>
      this article
    </p>
    <img
      alt="a title"
      src="that image url"
      style="height: 100px;"
    />
  </div>
</DocumentFragment>
`;
